wallet/src/repository/wallet/mod.rs

```diff
@@ -1203,9 +1230,16 @@ where
         utxo_strategy: &UtxoSelectionStrategy,
         max_weight: u32,
     ) -> Result<(Vec<Input>, Vec<ValueTransferOutput>)> {
+        let empty_hashset = HashSet::default();
+        let unconfirmed_transactions = if self.params.use_unconfirmed_utxos {
+            &empty_hashset
+        } else {
+            &state.pending_transactions
+        };
```

Comment:
Is it not the opposite logic? It means, if we want to use unconfirmed_utxos we include the pending transactions

Reply:
I didnt say nothing, it is to create the hashset to check if we can use or not, so it is ok

Reply:
It seems backwards but this is the logic: `unconfirmed_transactions` is the list of transactions that will not be used. If `unconfirmed_transactions` is empty, we will use all the unconfirmed UTXOs (this is the old behavior before this PR). If `unconfirmed_transactions` is equal to pending transactions, we will not use unconfirmed UTXOs in new transactions.